### DISCUSSION TITLE
Can I make the overlay non-interactive with the pointer?

### DISCUSSION THREAD

**Discussion by PremiumUsername** [wmww/gtk-layer-shell#150] [Q&A] [answered]
Say that you would like to make a semi opaque overlay over part of the screen in order to display some kind of information. How do you make it so it doesn't interact with the pointer, so that the pointer still can be used to control whatever is under the semi opaque widget.  

I'm using python. And the in my testing I've been using an  `Gtk.EventBox` that I put inside a window. Even if I make either of those  `insensitive', it will still steal the control over the pointer from whatever is under it. I've been playing with `set_device_events` and `set_events` as well, but I'm guessing that's filtering on the wrong level. Any pointers would be appreciated.

**Comment by ErikReider** [comment]
Have you tried enabling passthrough in your gdk window?

**Comment by PremiumUsername** [comment]
I'm very new at this. I created a window using `window = Gtk.Window()`.  I don't know how to use Gdk to create a window directly.

I now tried:
```
window = Gtk.Window()
window.get_root_window().set_pass_through(True)
```
It didn't solve my problem. Is it possible to enable passthrough that way?

**Comment by wmww** [comment, marked as answer]
Hmm, sounds like you might be running into #30. You might be able to solve this by manually setting an input region with Wayland, though that might be a bit tricky, especially from Python.

**Reply by wmww** [reply]
Update: #30 is actually fixed (has been for years, no one noticed until now). Therefore this should be easy. I tested the following in the python demo and it seems to work:
```python
import cairo
region = cairo.Region()
window.input_shape_combine_region(region)
```

**Reply by PremiumUsername** [reply]
> it seems to work

Yes, that works! Thank you! 

I didn't include this in my question but I was hoping that the would-be-solution would still allow me to use the touch screen. Would it be possible for the widget still be able to capture touch events, while pointer events would remain uncaptured? 

This is what I've tried:
```
import gi
import cairo

gi.require_version('Gtk', '3.0')
gi.require_version('GtkLayerShell', '0.1')

from gi.repository import Gtk,Gdk, GtkLayerShell

window = Gtk.Window()
Gtk.Widget.set_opacity(window, 0.5)

DISPLAY='wayland-1'
display = Gdk.Display().open(DISPLAY)
seat = display.get_default_seat()
pointer_device = Gdk.Seat.get_pointer(seat)
touch_device = Gdk.Seat.get_slaves(seat, Gdk.SeatCapabilities.TOUCH)[0]


def test_input(source, event):
    print(f'event: {event}')
    print(f'source: {source}')

GtkLayerShell.init_for_window(window)
GtkLayerShell.set_layer(window, GtkLayerShell.Layer.TOP)
GtkLayerShell.set_anchor(window, GtkLayerShell.Edge.RIGHT,1)

event_box = Gtk.EventBox()
event_box.set_size_request(100,800)
event_box.set_events(Gdk.EventMask.TOUCH_MASK)

# Doesn't seem to do anything
event_box.set_device_events(touch_device,Gdk.EventMask.TOUCH_MASK)

# Allow clickthrough for bottom half
region = cairo.Region(cairo.RectangleInt(0, 0, 100, 400))
window.input_shape_combine_region(region)

event_box.connect("event", test_input)

window.add(event_box)
window.show_all()

window.connect('destroy', Gtk.main_quit)
Gtk.main()
```

A lot of pointer events are still captured, so I don't think I understand what `set_device_events` is supposed to do, or I'm using it wrong.

**Reply by wmww** [reply]
The underlying Wayland protocol doesn't allow a window to intercept touch events in a particular position let pointer events pass through. In Wayland events always go to a single surface. Wayland surfaces can set their input shape, but both touch and pointer events respect the same input shape.

**Reply by PremiumUsername** [reply]
I see. Thank you for explaining this to me. I would probably have wasted a fair amount of time trying to achieve the impossible should you not have done me this favor.
